src/main/java/com/useful/ucars/uCarsListener.java

```diff
@@ -1286,10 +1278,10 @@ void safeFly(EntityDamageEvent event) {
 		}
 		Player p = (Player) event.getEntity();
 		if (inACar(p.getName())) {
-			Vector vel = p.getVehicle().getVelocity();
+			event.setCancelled(true);
+			/*Vector vel = p.getVehicle().getVelocity();
```

Comment:
The purpose of this code originally was to cancel fall damage only for small falls and otherwise leave it to other event handlers to deal with. If you want to customise it on your server to always cancel all fall damage then add your own event listener to do so instead of changing the behaviour of this one

Reply:
(or add a config option, default disabled, to remove all fall damage whilst in cars)

Reply:
Ah - The comment says "This disables fall damage whilst driving a car" so I figured it was supposed to disable all fall damage.

Interestingly enough the code was written... the wrong way around?
I don't remember if I changed that at some point for some reason but it would only serve minor so... non-existent fall damage.
There is a strange side-effect tho: You sometimes will be dealt fall damage whilst climbing up a for a longer distance (Y-Vel:-0.14).

I'll add a config-value and will try to get around that strange behaviour...
I made the plugin output my current fallDistance whenever it deals me damage and whilst climbing it has always been between 5-10 so maybe instead of going for Y-Speed go for fallDistance>10?

But I'll go dig through the code in order to find out why I'm suddenly "falling" whilst climbing... Maybe because I'm constantly "flying"?

Reply:
I think it sounds like it is mostly working as intended - going up/down slopes you are not getting fall damage (at least on earlier versions without it players when driving up/down hills could get fall damage with the velocities being applied to the car) and if you actually fall from a reasonable height you are. Fall distance I don't think works correctly on it's own for cars too - if the y offset the car is driving at is computed slightly off them the car may be driving ever so slightly above the ground for a while and then when it next 'hits the ground' again it can think it has fallen.

If the bounds needs to be enlargened to make sure when you are climbing your longer distance you are not getting fall damage then you can do so - perhaps something changed with the climbing code in your previous changes which means it is needed

Reply:
The goal isn't to make it so that a player can fall infinitely far in a car and survive, the goal is to make it so that (for the base uCars plugin) players aren't being damaged in normal driving up/down slopes etc...